src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -3977,7 +3981,7 @@ class StubGenerator: public StubCodeGenerator {
   void md5_FF(BufRegCache& reg_cache,
               Register a, Register b, Register c, Register d,
               int k, int s, int t,
-              Register rtmp1, Register rtmp2, Register rmask32) {
+              Register rtmp1, Register rtmp2) {
```

Comment:
Now that we have scratch register `t0` available for use in functions md5_FF/GG/HH/II, we could further remove use of `rtmp2`. Then we might let `rtmp1` = `t2` while use x10-x11 for state0-state1. This will help save use of one callee-saved register. Please consider.

Reply:
It was actually a conscious decision to use different registers (`a0`/`a1`) for the intermediate results in `md5_FF/GG/HH/II` than the registers (`t0` / `t1`) used in `m5_FF_GG_HH_II_epilogue`. Fewer register dependencies between instructions can maybe help some cpus. Also, `rtmp1` / `rtmp2` being `a0` / `a1` allows the use of compressed instructions in a couple of places (if we care about compressed instructions that is!).

Reply:
OK. That sounds like a good point. Thanks.